Cargo.toml

```diff
@@ -39,13 +39,20 @@ members = [
 # Ignore bindings stuff given they are special snowflakes that require exotic toolchain and dependencies
 default-members = ["oxidation/libparsec"]
 
-# A custom profile that is the fusion of `release` & `debug` profile.
-# We want the speed of the `release` profile but with the debug assertions of the `debug` profile.
-[profile.integration]
-inherits = "release"
-overflow-checks = true
-debug-assertions = true
-debug = true
-incremental = true
+# Rust unoptimized code is so slow that we customize dev profile to enable some
+# optimizations, otherwise Python tests take forever.
```

Comment:
Asking the dev to use `maturin develop` with `--profile=foo` is too much work ?

Reply:
Please refrain from passive-aggressive.

I agree we most likely want to use `opt-level = 0` for Rust dev and this addition probably solves as much as it breaks 😄 

We need a no-brainer way of building the project from Python, expecting user to know `maturin develop --profile=foo` not acceptable (having to use `maturin develop` instead of just `poetry install` is already annoying enough !)

I've reverted this change, we need more discussion on how to solve this

Reply:
I'll try my best :stuck_out_tongue: 

I've found that we can configure the profile used by `maturin`

https://www.maturin.rs/metadata.html#add-maturin-build-options

we can add

```toml
[tool.maturin]
profile = "ci-python" # or even release :thinking: 
```

Reply:
> I've found that we can configure the profile used by maturin in pyproject.toml

That's an interesting, however I think it will clash between release and dev build of the project (we want to use release profile for default build, and a custom python-dev profile when building with maturin develop),
I'll have a look at maturin to see if a different config can be applied between `maturn build` and `maturin develop`

Reply:
providing the arg `--profile` override the configured `profile` in the `pyproject.toml`.
In the `build.py` we provide the flag `--profile`